twine/commands/upload.py

```diff
@@ -48,6 +53,39 @@ def skip_upload(response, skip_existing, package):
             (response.status_code == 403 and msg_403 in response.text)))
 
 
+def check_status_code(response, verbose):
+    """
+    Additional safety net to catch response code 410 in case the
+    UploadToDeprecatedPyPIDetected exception breaks.
+    Also includes a check for response code 405 and prints helpful error
+    message guiding users to the right repository endpoints.
+    """
+    if (response.status_code == 410 and
```

Comment:
Would you be game to add a test for this as well? I realize it's not *new* code, but since it moved, I think it's reasonable to add test coverage. I think it would also make codecov happy.

Reply:
I thought of that, but the previous docstring indicates that this is a second line of defense, and that `UploadToDeprecatedPyPIDetected` is raised earlier. This is tested: https://github.com/pypa/twine/blob/master/tests/test_upload.py#L110-L126

We can, however, add a test that calls `check_status_code` directly with the appropriate arguments and check if it works okay.

Reply:
Good point. That said, I think as long as this post-upload condition is there, it's worth testing.

Reply:
Okay, I'll add a test that calls `check_status_code` directly.

Reply:
Added in 9068268.